apps/aecontract/src/aect_contracts.erl

```diff
@@ -240,6 +246,15 @@ new(RTx) ->
 -spec new(aec_keys:pubkey(), ct_nonce(), version(), binary(), amount()) -> contract().
 %% NOTE: Should only be used for contract execution without transaction
 new(Owner, Nonce, CTVersion, Code, Deposit) ->
+    new_internal(Owner, Nonce, CTVersion, {code, Code}, Deposit).
+
+%% NOTE: When cloning passed CodeRefPK has to have code_ref=no_ref (NO INDIRECT CODE REFERENCE)
+-spec new_clone(aec_keys:pubkey(), ct_nonce(), version(), aec_keys:pubkey(), amount()) -> contract().
+new_clone(Owner, Nonce, CTVersion, CodeRefPK, Deposit) ->
+    CodeRef = aeser_id:create(contract, CodeRefPK),
```

Comment:
Why do we use a full id here? Why not just the bytes?

Reply:
Cause we have a serialization type for IDs and not for fixed size byte fields. Also we use IDs for other simillar fields like the `id` field.